docs/api-reference/mapbox/overview.md

```diff
@@ -33,29 +35,42 @@ import {MapboxOverlay} from '@deck.gl/mapbox';
 
 It may be easier to understand the concepts of the module if you are already a mapbox-gl developer.
 
+### Using mapbox-gl controls with deck.gl
+
+The Mapbox ecosystem offers many well-designed controls, from the basic functionalities of `NavigationControl`, `Popup` and `GeolocateControl`, to vendor-service-bound UI implementations such as `mapbox-gl-geocoder` and `mapbox-gl-directions`. These libraries require that the Mapbox Map holds the source of truth of the camera state, instead of the normal [state management](../../developer-guide/interactivity.md) by `Deck`. When you use the `MapboxOverlay` or `MapboxLayer` classes from this module, deck.gl plays nice with all the mapbox-gl peripherals.
+
+
 ### Mixing deck.gl layers and Mapbox layers
 
 One major use case for interleaving deck.gl and Mapbox is that some important information in the Mapbox map could be hidden by a deck.gl visualization layer, and controlling opacity is not enough. A typical example of this is labels and roads, where it is desirable to have a deck.gl visualization layer render on top of the Mapbox geography, but where one might still want to see e.g. labels and/or roads. Alternatively, the deck.gl visualization should cover the ground, but not the roads and labels.
 
 To inject a deck layer into the Mapbox stack, either:
 
+- Add a `beforeId` prop to any layer passed to the [MapboxOverlay](./mapbox-overlay.md) control (recommended).
 - Create a [MapboxLayer](./mapbox-layer.md) and call the [`map.addLayer(layer, before?)`](https://www.mapbox.com/mapbox-gl-js/api/#map#addlayer) API.
-- Add a `beforeId` prop to any layer passed to the [MapboxOverlay](./mapbox-overlay.md) control.
 
 Mapbox provides an example of [finding the first label layer](https://www.mapbox.com/mapbox-gl-js/example/geojson-layer-in-stack/). For more sophisticated injection point lookups, refer to Mapbox' documentation on the format of Mapbox style layers, see [Mapbox Style Spec](https://www.mapbox.com/mapbox-gl-js/style-spec/#layers).
 
 
 In some cases, the application wants to add a deck.gl 3D layer (e.g. ArcLayer, HexagonLayer, GeoJsonLayer) on top of a Mapbox basemap, while seamlessly blend into the z-buffer. This will interleave the useful visualization layers from both the deck.gl and Mapbox layer catalogs. In this case, a `beforeId` is not needed.
 
+## Compatibility
```

Comment:
I looked for the first release the WebGL2 compatibility commit appeared to make this table:
- Mapbox `useWebGl` flag: https://github.com/mapbox/mapbox-gl-js/commit/d7652ba8eda70a97809840364beed81bad07930d
- Mapbox by default: https://github.com/mapbox/mapbox-gl-js/commit/27aa25082af00c83c25e92bff87c001c0137aafd
- Maplibre by default: https://github.com/maplibre/maplibre-gl-js/commit/7a1d5b5355f9b1da31ced9d6f1731b9828f8e057